wear/src/main/java/io/homeassistant/companion/android/sensors/HealthServicesSensorManager.kt

```diff
@@ -175,25 +224,14 @@ class HealthServicesSensorManager : SensorManager {
                         }
                     }
                 }
-                if (floorsDaily.isNotEmpty()) {
-                    floorsDaily.forEachIndexed { index, intervalDataPoint ->
-                        val endTime = intervalDataPoint.getEndInstant(bootInstant)
-                        Log.d(TAG, "Daily Floors data index: $index with value: ${intervalDataPoint.value} end time: ${endTime.toEpochMilli()}")
-                        if (endTime.toEpochMilli() > latest) {
-                            latest = endTime.toEpochMilli().toInt()
-                            lastIndex = index
-                        }
-                    }
-                    onSensorUpdated(
-                        latestContext,
-                        dailyFloors,
-                        floorsDaily[lastIndex].value,
-                        dailyFloors.statelessIcon,
-                        mapOf()
-                    )
 
+                if (
+                    processDataPoint(floorsDaily, dailyFloors) ||
+                    processDataPoint(distanceDaily, dailyDistance) ||
+                    processDataPoint(caloriesDaily, dailyCalories) ||
+                    processDataPoint(stepsDaily, dailySteps)

```

Comment:
Stacking the processing functions like this means that if the app receives multiple data point types at once, other data points aren't processed: it will stop executing functions once one of these returns `true`. For example, if the app were to receive distance and steps at the same time, only the distance data points would be processed.

Reply:
Ah I probably didn't catch this in testing as I only got 1 data type per update. I'll submit a change shortly so we can ensure each data point is processed before sending the update.

Reply:
I don't know if it actually happens but think the app should handle it just in case sensor updates are batched

Reply:
yea they could be batched if we decide to switch from the callback to the service. I went with the callback only because its said to be faster.

`You can receive passive data through a service or a callback (or both). A service allows your app to receive data in the background when no part of your app is visible in the foreground. When you receive data in the background it is delivered in batches. The callback receives data at a slightly faster rate, but only while the app is running and the callback is successfully notified.`

as we are a foreground service the callback works just fine :) except for when we are hit by doze lol but thats out of our control

Reply:
> Also the app isn't using a foreground service all the time for the callback so it is probably still "in the background" as far as Android is concerned.

i havent seen any issues with using the callback, updates still come in as expected throughout the day.